pkg/api/v1/atlasdatabaseuser_types.go

```diff
@@ -73,10 +73,13 @@ type AtlasDatabaseUserSpec struct {
 	Scopes []ScopeSpec `json:"scopes,omitempty"`
 
 	// PasswordSecret is a reference to the Secret keeping the user password.
-	PasswordSecret *ResourceRef `json:"passwordSecretRef"`
+	PasswordSecret *ResourceRef `json:"passwordSecretRef,omitempty"`
 
 	// Username is a username for authenticating to MongoDB.
 	Username string `json:"username"`
+
+	// X509Type is X.509 method by which the database authenticates the provided username
+	X509Type string `json:"x509Type,omitempty"`
```

Comment:
I think explicit X.509 is better but. `X509Type` is an actual field of the resulting request and it can be something other than `CUSTOMER` (so we just exposing this field to the user), this way it's easier to follow from the existing Atlas documentation. At the same time, I will think again about it (maybe there is actually a better solution, like you've suggested).